src/main/scala/com/microsoft/hyperspace/actions/RefreshActionBase.scala

```diff
@@ -71,15 +71,16 @@ private[actions] abstract class RefreshActionBase(
   // Reconstruct a df from schema
   protected lazy val df = {
     val relations = previousIndexLogEntry.relations
-    val latestRelation = Hyperspace
+    val relationMetadata = Hyperspace
       .getContext(spark)
       .sourceProviderManager
       .getRelationMetadata(relations.head)
-      .refresh()
+    val latestRelation = relationMetadata.refresh()
     val dataSchema = DataType.fromJson(latestRelation.dataSchemaJson).asInstanceOf[StructType]
-    val df = spark.read
-      .schema(dataSchema)
-      .format(latestRelation.fileFormat)
+    val df = {
+      if (relationMetadata.canSupportUserSpecifiedSchema) spark.read.schema(dataSchema)
```

Comment:
oh, is this fixing an existing bug?

Reply:
Maybe they were ignored in Spark 2. In Spark 3 it throws an exception if schema is specified for Delta or Iceberg relations.